### PR TITLE
Idle time support

### PR DESCRIPTION
**Pull request by mdekstrand**
This adds support for 'natural breaks' using [system-idle-time](https://github.com/paulcbetts/node-system-idle-time), working on #59.

I have only tested this code on Windows.

It adds one settting to the settings screen, but I have not updated the screenshot because Windows will be inconsistent with Mac.

### CODE REVIEW

**Comment by sxmanton** [comment]
Hi, I'm trying to build this and give it a test, however, when I run npm install, I'm getting an error: "paulcbetts/system-idle-time is not in the npm registry". Any ideas what the issue might be?

**Comment by sxmanton** [comment]
Nevermind, I just figured out that somehow I had two different versions of npm installed and one was much lower version. Uninstalled the lower version and it got past that step, although it still failed with "@paulcbetts/system-idle-time@1.04 install: 'node-gyp rebuild' Exit status 1". This issue seems to be referenced here: https://github.com/RocketChat/Rocket.Chat.Electron/issues/335 so I'm now trying some of those suggested steps. Did you have to go through such a process at all? Might be worth adding something to the readme about

File: app/breaksPlanner.js [hunk 1]
```diff
@@ -15,17 +20,31 @@ class BreaksPlanner {
     let interval = this.settings.get('microbreakInterval')
     if (!shouldBreak && shouldMicrobreak) {
       this.scheduler = new Scheduler(this.microbreakFunc, interval)
+      this.nextBreakDuration = this.settings.get('microbreakDuration')
     } else if (shouldBreak && !shouldMicrobreak) {
       this.scheduler = new Scheduler(this.breakFunc, interval * (this.settings.get('breakInterval') + 1))
+      this.nextBreakDuration = settings.get('breakDuration')
```
Comment (sxmanton) [inline app/breaksPlanner.js]: Should be this.settings instead of settings

File: app/breaksPlanner.js [hunk 2]
```diff
@@ -15,17 +20,31 @@ class BreaksPlanner {
...
     } else if (shouldBreak && shouldMicrobreak) {
       this.breakNumber = this.breakNumber + 1
       let breakInterval = this.settings.get('breakInterval') + 1
       if (this.breakNumber % breakInterval === 0) {
         this.scheduler = new Scheduler(this.breakFunc, interval)
+        this.nextBreakDuration = settings.get('breakDuration')
```
Comment (sxmanton) [inline app/breaksPlanner.js]: Should be this.settings instead of settings

**Comment by mdekstrand** [comment]
I didn't have any build failures, but did need to use `electron-rebuild` to avoid a load failure.

On Linux it will require the XScreenSaver extension dev files (Ubuntu package `libxss-dev`).

**Comment by hovancik** [comment]
I've updated this with master. No idea if it works, thought.

**Comment by hovancik** [comment]
@mdekstrand could you please explain how your mechanism works? I have hard times understanding it.

**Comment by hovancik** [comment]
seems that author is no longer interested in this so I am closing it. I'm working on new solution at #140

**Comment by mdekstrand** [comment]
fine with me, i haven't had time to look at it any more since submitting :(

**Pull request closed without merging**